javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -22,15 +24,19 @@
  * @author Ruslan Sennov
  * @since 2.0.0
  */
-public final class Vector<T> implements Kind1<Vector<?>, T>, IndexedSeq<T>, Serializable {
-
+@SuppressWarnings({"SuspiciousArrayCast", "unchecked"})
```

Comment:
I would revert this back. It should be done deliberately, in place of code where it is needed. Somebody can make a mistake in the future because warnings switched off in whole class